website/docs/quick-start-guide.md

```diff
@@ -803,30 +859,37 @@ when not matched then
 
 ```python
 # pyspark
+snapshotBeforeUpdate = spark.sql(snapshotQuery)
 updates = sc._jvm.org.apache.hudi.QuickstartUtils.convertToStringList(dataGen.generateUpdates(10))
 df = spark.read.json(spark.sparkContext.parallelize(updates, 2))
-df.write.format("hudi"). \
-  options(**hudi_options). \
-  mode("append"). \
-  save(basePath)
+df.write.format("hudi").
+options(**hudi_options).
+mode("append").
+save(basePath)
+assert spark.sql(snapshotQuery).intersect(df).count() == df.count()
```

Comment:
lets assign spark.sql(snapshotQuery) to another variable and cache it. if not, we might keep retriggering the actions.